lib/EventListener/UserEditListener.php

```diff
@@ -9,10 +9,15 @@
 use eZ\Publish\API\Repository\ContentTypeService;
 use eZ\Publish\API\Repository\Exceptions\NotFoundException;
 use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\Core\Repository\Repository;
```

Comment:
Using Core Repository will lead to bypassing all the layers. This can't be done in an external package.

Reply:
Undestood, but I strictly need sudo method.

Reply:
They are there, you just can't type hint it as it's on purpose left out from interface.

Plan was to split permissions to own layer, but also multi lookup api calls [rfc](https://github.com/ezsystems/ezpublish-kernel/pull/2070) would allow you to read data you don't have access to.

So please type hint API even if it lacks sudo.